sql/core/src/test/scala/org/apache/spark/sql/DatasetCacheSuite.scala

```diff
@@ -21,11 +21,32 @@ import scala.language.postfixOps
 
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.test.SharedSQLContext
+import org.apache.spark.storage.StorageLevel
 
 
 class DatasetCacheSuite extends QueryTest with SharedSQLContext {
   import testImplicits._
 
+  test("get storage level") {
+    val ds1 = Seq("1", "2").toDS().as("a")
+    val ds2 = Seq(2, 3).toDS().as("b")
+
+    // default storage level
+    ds1.persist()
+    ds2.cache()
+    assert(ds1.storageLevel() == StorageLevel.MEMORY_AND_DISK)
+    assert(ds2.storageLevel() == StorageLevel.MEMORY_AND_DISK)
+    // unpersist
+    ds1.unpersist()
+    assert(ds1.storageLevel() == StorageLevel.NONE)
+    // non-default storage level
+    ds1.persist(StorageLevel.MEMORY_ONLY_2)
```

Comment:
When writing black-box testing, I might just try all the levels in the test case, even we can include some customized StorageLevel, which is diffrent from the defined one.
```scala
    Seq(StorageLevel.OFF_HEAP, MEMORY_AND_DISK_SER_2, ..., StorageLevel.NONE).foreach { level =>
      ds1.persist(level)
      assert(ds1.storageLevel() == level)
      ds1.unpersist()
      assert(ds1.storageLevel() == StorageLevel.NONE)
    }
```